src/data_readers/data_reader_imagenet.cpp

```diff
@@ -43,17 +43,17 @@ imagenet_reader::imagenet_reader(const std::shared_ptr<cv_process>& pp, bool shu
     throw lbann_exception(err.str());
   }
 
-  replicate_processor(*pp);
+  m_master_pps.reset(new cv_process(*pp));
 }
 
 imagenet_reader::imagenet_reader(const imagenet_reader& rhs)
   : image_data_reader(rhs) {
-  if (rhs.m_pps.size() == 0u || !rhs.m_pps[0]) {
+  if (!rhs.m_master_pps) {
     std::stringstream err;
     err << __FILE__<<" "<<__LINE__<< " :: " << get_type() << " construction error: no image processor";
     throw lbann_exception(err.str());
   }
-  replicate_processor(*rhs.m_pps[0]);
+  m_master_pps.reset(new cv_process(*rhs.m_master_pps));
 }
 
 imagenet_reader& imagenet_reader::operator=(const imagenet_reader& rhs) {
```

Comment:
Somewhat outside the scope of this PR, but since I'm noticing the terribly duplicated code below: http://gotw.ca/gotw/059.htm @timmoon10